### DISCUSSION TITLE
Make windows float if it is non-resizable

### DISCUSSION THREAD

**Discussion by Menchen** [asmvik/yabai#720] [Q&A]
It would help a lot if there's an option that automatically makes windows float if it's non-resizable or a new signal...

I tried using signal but it seems that some apps change to non-resizable after its creation (Intellij IDEA, the welcome screen) and one way I think I can do it is polling all windows with `yabai -m query`, but it's so slow and adds overheat to the system, maybe a signal for windows property change?

Another example is Finder copy progression windows... it would mess up my current space and allocate a big chunk of space just for a tiny window.

**Comment by asmvik** [comment]
This used to be the default, but had to change for reasons that I no longer remember.
You can identify patterns for these windows and create rules that automatically make them float.

e.g: `yabai -m rule --add app="^Steam$" manage=off`
See https://github.com/koekeishiya/yabai/blob/master/doc/yabai.asciidoc#rule

**Comment by anujc4** [comment]
You could use a similar configuration
```
# Do not manage windows with certain titles eg. Copying files or moving to bin
yabai -m rule --add title="(Copy|Bin|About This Mac|Info)" manage=off

# Do not manage some apps which are not resizable
yabai -m rule --add app="^(Calculator|VLC|System Preferences)$" manage=off
```

**Comment by Inc0n** [comment]
I would like to suggest to bring back the unmanage all unresizable windows as an option in configuration.
I have used a signal to float the window on creation, however, there are two small issues here:
1. If my current window is zoom-fullscreened, and i for example trigger this type of window (download a file), it manages this window, and destroyes my current space configuration.
2. Using the single approach is slow, and it can be visibly seen that the window is floatified, and to me is considered unpleasant, or that I would prefer no resizing animation.

On top of this, these download popup windows do not have a consistent title name, thus could not be rule managed. Then using singles brings us back to the two points mentioned above.

**Comment by landoncrabtree** [comment]
Bump on this.. Things like Finder copying files, moving files, throws my windows around and the popup has no title so I can't create rules to manage it.

**Reply by Bellavene** [reply]
```
yabai -m rule --add \
         label="Finder floating" \
         app="^Finder$" \
         title="(Copy|Connect|Move|Info|Finder Settings)" \
         manage=off

```
